prov/efa/src/efa_prov_info.c

```diff
@@ -158,37 +164,33 @@ const struct fi_ep_attr efa_ep_attr = {
  *
  * @param	prov_info[out]		prov_info pointer
  * @param	device[in]		pointer to an efa_device struct, which contains device attributes
- * @param	ep_type			endpoint_type, can be FI_EP_RDM or FI_EP_DGRAM
+ * @param	info_type			EFA info type
  */
 static
 void efa_prov_info_set_ep_attr(struct fi_info *prov_info,
 			       struct efa_device *device,
-			       enum fi_ep_type ep_type)
+			       enum efa_info_type info_type)
 {
 
 	*prov_info->ep_attr = efa_ep_attr;
-	if (ep_type == FI_EP_DGRAM) {
-		prov_info->mode |= FI_MSG_PREFIX;
-		prov_info->ep_attr->msg_prefix_size = 40;
-	}
-
 	prov_info->ep_attr->protocol	= FI_PROTO_EFA;
-	prov_info->ep_attr->type	= ep_type;
-
-	if (prov_info->ep_attr->type == FI_EP_RDM) {
-		prov_info->tx_attr->inject_size = device->efa_attr.inline_buf_size;
-	} else {
-		assert(prov_info->ep_attr->type == FI_EP_DGRAM);
-                /*
-		 * Currently, there is no mechanism for device to discard
-		 * a completion, therefore there is no way for dgram endpoint
-		 * to implement FI_INJECT. Because FI_INJECT is not an optional
-		 * feature, we had to set inject_size to 0.
-		 * 
-		 * TODO:
-		 * Remove this after implementing cq read for efa-raw
-                 */
-		prov_info->tx_attr->inject_size = 0;
+	prov_info->tx_attr->inject_size = device->efa_attr.inline_buf_size;
+
+	switch (info_type) {
+		case EFA_INFO_RDM:
+			prov_info->ep_attr->type = FI_EP_RDM;
+			break;
+		case EFA_INFO_DIRECT:
+			prov_info->ep_attr->type = FI_EP_RDM;
+			prov_info->mode |= FI_CONTEXT2; 	/* EFA direct provider requires FI_CONTEXT2 mode */
+			break;
+		case EFA_INFO_DGRAM:
```

Comment:
dgram should require FI_CONTEXT2 now as well. as we have the same efa-direct code for dgram

Reply:
done